integration_tests/src/main/python/rapids_udf_test.py

```diff
@@ -21,104 +26,48 @@
 from pyspark.sql.utils import AnalysisException
 from conftest import skip_unless_precommit_tests
 
-encoded_url_gen = StringGen('([^%]{0,1}(%[0-9A-F][0-9A-F]){0,1}){0,30}')
-
 def drop_udf(spark, udfname):
     spark.sql("DROP TEMPORARY FUNCTION IF EXISTS {}".format(udfname))
 
 def skip_if_no_hive(spark):
     if spark.conf.get("spark.sql.catalogImplementation") != "hive":
         skip_unless_precommit_tests('The Spark session does not have Hive support')
 
-def load_hive_udf_or_skip_test(spark, udfname, udfclass):
+def load_hive_udf(spark, udfname, udfclass):
     drop_udf(spark, udfname)
     try:
         spark.sql("CREATE TEMPORARY FUNCTION {} AS '{}'".format(udfname, udfclass))
-    except AnalysisException:
-        skip_unless_precommit_tests("UDF {} failed to load, udf-examples jar is probably missing".format(udfname))
-
-def test_hive_simple_udf():
-    with_spark_session(skip_if_no_hive)
-    data_gens = [["i", int_gen], ["s", encoded_url_gen]]
-    def evalfn(spark):
-        load_hive_udf_or_skip_test(spark, "urldecode", "com.nvidia.spark.rapids.udf.hive.URLDecode")
-        return gen_df(spark, data_gens)
-    assert_gpu_and_cpu_are_equal_sql(
-        evalfn,
-        "hive_simple_udf_test_table",
-        "SELECT i, urldecode(s) FROM hive_simple_udf_test_table")
-
-def test_hive_generic_udf():
-    with_spark_session(skip_if_no_hive)
-    def evalfn(spark):
-        load_hive_udf_or_skip_test(spark, "urlencode", "com.nvidia.spark.rapids.udf.hive.URLEncode")
-        return gen_df(spark, [["s", StringGen('.{0,30}')]])
-    assert_gpu_and_cpu_are_equal_sql(
-        evalfn,
-        "hive_generic_udf_test_table",
-        "SELECT urlencode(s) FROM hive_generic_udf_test_table")
-
-    def evalfn_decimal(spark):
-        load_hive_udf_or_skip_test(spark, "fraction", "com.nvidia.spark.rapids.udf.hive.DecimalFraction")
-        return gen_df(spark, [["dec", DecimalGen(38, 18)]])
-    assert_gpu_and_cpu_are_equal_sql(
-        evalfn_decimal,
-        "hive_generic_udf_test_table",
-        "SELECT fraction(dec) FROM hive_generic_udf_test_table")
+    except AnalysisException as e:
+        # rethrow this error, or test case will be skipped silently
+        raise RuntimeError("UDF {} failed to load, check if {} is in the class path, error is {}".format(udfname, udfclass, repr(e)))
```

Comment:
Curious, why not let the original exception be thrown directly?  Won't it be obvious from the original exception which UDF class cannot be found in the classpath?

Reply:
Updated